### PR TITLE
Point Getting Started to docker-compose

### PR DESCRIPTION
**Pull request by mcdan**
Updates the readme.md to point new people encountering OW to the docker-compose tooling.

## Description

Change some links add how to get the docker-compose stuff running quickly.

## Related issue and scope

Discussed a fair bit in the slack channel ~ https://openwhisk-team.slack.com/archives/C3TPCAQG1/p1523544305000779

## My changes affect the following components

- [x] Documentation
- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [ ] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [ ] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

**Comment by ddragosd** [comment]
LGTM.  We can decide later about the state of Vagrant.

**Comment by chetanmeh** [comment]
Would be helpful to also validate if docker compose based setup works fine on Windows or not and see what minimum Windows version is required for this to work

**Comment by mcdan** [comment]
@chetanmeh currently getting a windows VM running so I can check on that for you.

File: README.md [hunk 1]
```diff
@@ -25,15 +25,27 @@
 OpenWhisk is a cloud-first distributed event-based programming service. It provides a programming model to upload event handlers to a cloud service, and register the handlers to respond to various events. Learn more at [http://openwhisk.incubator.apache.org](http://openwhisk.incubator.apache.org).
 
 
-* [Quick Start](#quick-start) (Vagrant)
+* [Quick Start](#quick-start) (Docker-Compose)
+* [Vagrant](#vagrant-setup)
```
Comment (csantanapr) [inline README.md]: Move Vagrant down after native
Add (Windows) [Vagrant](#vagrant-setup) (Windows)

File: README.md [hunk 2]
```diff
@@ -25,15 +25,27 @@
 OpenWhisk is a cloud-first distributed event-based programming service. It provides a programming model to upload event handlers to a cloud service, and register the handlers to respond to various events. Learn more at [http://openwhisk.incubator.apache.org](http://openwhisk.incubator.apache.org).
```
Comment (csantanapr) [inline README.md]: Add (Mac and Linux) to quick start line

File: mesos.env [hunk 3]
```diff
@@ -0,0 +1 @@
+CONFIG_whisk_mesos_master-url = "http://m1.dcos"
```
Comment (csantanapr) [inline mesos.env]: what's this?

File: .project [hunk 4]
```diff
@@ -5,6 +5,11 @@
 	<projects>
 	</projects>
 	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
```
Comment (csantanapr) [inline .project]: Are you one of those that do `git add .` ? 😸

Comment (mcdan) [reply .project]: Did your update on a different machine, which apparently doesn't have the same git global config, not sure why that sync is broken.

**Comment by ddragosd** [comment]
Is there a pending action item here or we can merge this ?

**Comment by rabbah** [comment]
On a fresh system, got one error and some warnings:

```
cat: /Users/rabbah/tmp/openwhisk/local.env: No such file or directory
...
 [WARNING]: Unable to parse /openwhisk/ansible/environments/local as an inventory source
 [WARNING]: No inventory was parsed, only implicit localhost is available
 [WARNING]: Could not match supplied host pattern, ignoring: all
 [WARNING]: provided hosts list is empty, only localhost is available
```

system came up cleanly though:
```
creating the hello.js function ...
invoking the hello-world function ... 
adding the function to whisk ...
ok: created action hello
invoking the function ...
invokation result: { "payload": "Hello, World!" }
{ "payload": "Hello, World!" }
 creating an API from the hello function ...
ok: updated action hello
invoking:  http://172.17.0.1:9090/api/23bc46b1-71f6-4ed5-8c54-816aa4f8c502/hello/world
  "payload": "Hello, World!"
ok: APIs
Action          Verb  API Name  URL
/guest/hello     get    /hello  http://172.17.0.1:9090/api/23bc46b1-71f6-4ed5-8c54-816aa4f8c502/hello/world
deleting the function ...
ok: deleted action hello
```

**Comment by ddragosd** [comment]
@rabbah  thanks for testing on a fresh system. The `WARNING` messages from Ansible should be fine b.c ansible usage is limited to configuring the DB. We may consider in the future to build an OW DB image at build time, push it to DockerHub, so that new deployments don't have to re-initialize the DB each time.

**Pull request merged**